File: bookstore-frontend/src/types/Book.ts
export interface Book {
  bookId: number;
  title: string;
  author: string;
  publisher: string;
  isbn: string;
  classification: string;
  category: string;
  numPages: number;
  price: number;
}

export interface BookResponse {
  books: Book[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

File: BookstoreAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookstoreAPI.Models
{
    public class Book
    {
        [Key]
        [Column("BookID")]
        public int BookId { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Author { get; set; } = string.Empty;

        [Required]
        public string Publisher { get; set; } = string.Empty;

        [Required]
        public string ISBN { get; set; } = string.Empty;

        [Required]
        public string Classification { get; set; } = string.Empty;

        [Required]
        public string Category { get; set; } = string.Empty;

        [Required]
        [Column("PageCount")]
        public int NumPages { get; set; }

        [Required]
        public decimal Price { get; set; }
    }
}
